### ISSUE TITLE
Please Add MongoDB in Appwrite

### ISSUE THREAD

**Issue by rupamking1** [appwrite/appwrite#909]
Please Add MongoDB in Appwrite

**Comment by noviicee** [comment]
Can you provide little more details?

**Comment by rupamking1** [comment]
MongoDB is a popular database

**Comment by eldadfux** [comment]
Probably still early, but we have started researching around the possibly of MongoDB support 👍🏻

**Comment by DevDorrejo** [comment]
Hello, yeah i want mongoDB too.

**Comment by kodumbeats** [comment]
Could you share how you'd like MongoDB integration to look @DevDorrejo? Our current database API provides a document-based approach, so would you prefer to use MongoDB in place of our current database?

**Comment by DevDorrejo** [comment]
> Could you share how you'd like MongoDB integration to look @DevDorrejo? Our current database API provides a document-based approach, so would you prefer to use MongoDB in place of our current database?

Hello, the second one, because my appwrite instance is not have own machine, and want to at least have none extra "RDBMS" packages with my instance installation.

now if this is possible, if no, all good too, can't ask too much for this great and complete project.

**Comment by alexweininger** [comment]
I think if it’s possible, it might be best to easily allow for people to use any database with AppWrite. It would be very nice to be able to select a database adapter (or write your own) and then just plug AppWrite into any kind of database, SQL, Mongo, etc

**Comment by scottie** [comment]
After playing around with appwrite I find it is perfect apart from this one issue.

**Comment by 6lackknight** [comment]
Yeah, allowing users to select their own databases makes it easy to make the decision to move to Appwrite

**Comment by ArpanKIIT2017** [comment]
Do we have any update on this?

**Comment by eabassey** [comment]
Is this still a priority?

**Comment by abhaysingh-info** [comment]
@eabassey as tag suggest it's work-in-progress we can just hope that it's integrated soon 🔥

**Comment by mikdanjey** [comment]
I want also mongodb database support in AppWrite

I have Serverless mongodb - https://www.mongodb.com/pricing

**Comment by achamas-playco** [comment]
Please add true JSON nosql support to the database capabilities.

As far as I understand the current MariaDB adapter requires tabular-style schemas, where each attribute is like a table column, with simple value types as data. What seems to be missing is the ability to define an attribute as an object/map, and then mark any attribute value as that object/map type. That way you could define nested hierarchical JSON structures.

Consider this structure for a hypothetical project document:

```json
{
   "id": "my-project",
   "nodes: {
      "node1": {
         ...node1 properties
      },
      "node2": {
         ...node1 properties
      }
   }
}
```

Each node is a schema type of its own, and may have sub schema types. We only want to store a single document for the project, but we want to be able to add/remove/modify nodes.

Technically this would be possible using the relational-style, where each node property would be a document of its own, but this is not practical or scalable as we would need to join and lookup all the nodes for a given project, effectively having a giant nodes collection. It would be worse if the nodes used subtypes as well.

Perhaps the solution would be to define an attribute type as another collection type, essentially storing the id of a document from that collection as the attribute value? That could handle the storage design but we would then need to be able to fetch the whole joined document, and write it back as a whole and have the sub types broken down and stored in their collections under the hood. Not sure how easy that would be to implement though.

Just a suggestion, but I would really love to see this kind of deep JSON-like feature either with a MongoDB adapter or some extension to the existing MariaDB feature. Feels like a crucial missing detail for a nosql database.

Thank you!

**Comment by mkhtradm01** [comment]
@adityaoberai you could use ORM like [Prisma](https://www.prisma.io) for this issue. It would make things easier, though I am a not PHP Developer and this mainly use typescript.

Cc: @alexweininger @achamas-playco

**Comment by AidenY69** [comment]
Any updates on MongoDB integration? Or providing some guidance on how to get it done via Functions?

**Comment by unknoxn** [comment]
Any updates @adityaoberai ?

**Comment by AidenY69** [comment]
@stnguyen90 was it removed from WIP because it's finished or because it's not happening?

**Comment by stnguyen90** [comment]
> was it removed from WIP because it's finished or because it's not happening?

@AidenY69, I'm deleting the "work in progress" label. We're still planning to provide some NoSQL support. For now, we want to stick with a officially supporting mariadb for easier maintenance and support.

**Comment by thezachdrake** [comment]
@stnguyen90 I see this ticket is on the board for the 1.7 release. Is that still the case or has this been postponed further out?

**Comment by 6lackknight** [comment]
> @stnguyen90 I see this ticket is on the board for the 1.7 release. Is that still the case or has this been postponed further out?

I hope you guys realize how many projects would benefit from this.

**Comment by benan789** [comment]
@stnguyen90 I feel like if given the choice between mariadb and mongodb more people would opt for mongodb. Also, with the deprecation of realm/mongo atlas services their customers will be looking for replacement solutions. This can be a huge opportunity if appwrite can provide a path for a easy migration where users can just plug in their existing mongodb. And as a side note, 5 functions for hobby seems way too low, supabase provides 25.

**Comment by polbuhanky** [comment]
It would be great to use combined solutions. For example, some collections can be created on maria, and others on mongodb.

**Comment by felixrydberg** [comment]
Would really like to see this intoduced. Easier data replication between multiple nodes to give your project redundancy
